### PR TITLE
K-Nunmpad17 

### PR DESCRIPTION
**Pull request by Karlssn**
I created a new handwired keyboard and added it under keyboards/handwired/ 

## Description
So I recently created a numpad and handwired it on my own but found that qmk didnt have a numpad with 17 keys under the handwired tab. Therefore I created my own based of Tritium Numpad which is a 21 key numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/k_numpad17/k_numpad17.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   * |----|----|----|----|
+   * | 30 | 31 | 32 |    |
+   * |----|----|----| 33 |
+   * |   40    | 41 |    |
+   * `-------------------'
+ */
+
+
+void keyboard_pre_init_user(void);
+void matrix_scan_user(void);
```
Comment (drashna) [inline keyboards/handwired/k_numpad17/k_numpad17.h]: These shouldn't be needed, actually.

File: keyboards/handwired/k_numpad17/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   * |----|----|----| +  |
+   * | 4  | 5  | 6  |    |
+   * |----|----|----|----|
+   * | 1  | 2  | 3  |    |
+   * |----|----|----| En |
+   * |    0    | .  |    |
+   * `-------------------'
+   */
+
+  [0] = layout_5x4(
```
Comment (drashna) [inline keyboards/handwired/k_numpad17/keymaps/default/keymap.c]:
```suggestion
  [0] = LAYOUT_numpad_5x4(
```

File: keyboards/handwired/k_numpad17/keymaps/karlssn/keymap.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+   * |----|----|----| +  |
+   * | 4  | 5  | 6  |    |
+   * |----|----|----|----|
+   * | 1  | 2  | 3  |    |
+   * |----|----|----| En |
+   * |   FN/0  | .  |    |
+   * `-------------------'
+   */
+
+  [0] = layout_5x4(
```
Comment (drashna) [inline keyboards/handwired/k_numpad17/keymaps/karlssn/keymap.c]:
```suggestion
  [0] = LAYOUT_numpad_5x4(
```

File: keyboards/handwired/k_numpad17/keymaps/karlssn/keymap.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   * | F7 | F8 | F9 |    |
+   * |----|----|----| vol|
+   * | F4 | F5 | F6 |  + |
+   * |----|----|----|----|
+   * | F1 | F2 | F3 | vol|
+   * |----|----|----|  - |
+   * |    FN   | .  |    |
+   * `-------------------'
+   */
+  [1] = layout_5x4(
```
Comment (drashna) [inline keyboards/handwired/k_numpad17/keymaps/karlssn/keymap.c]:
```suggestion
  [1] = LAYOUT_numpad_5x4(
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality (+4870)
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality (+1150)
+MIDI_ENABLE = no 		# MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH_ENABLE = no # Enable Bluetooth with the Adafruit EZ-Key HID
+TAP_DANCE_ENABLE = yes
+
+LAYOUTS = layout_5x4
```
Comment (drashna) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
LAYOUTS = numpad_5x4
```

File: keyboards/handwired/k_numpad17/readme.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+K-Numpad17
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/readme.md]:
```suggestion
# K-Numpad17
```

File: keyboards/handwired/k_numpad17/readme.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+K-Numpad17 
+===
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/readme.md]:
```suggestion

```

File: keyboards/handwired/k_numpad17/readme.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+| qmk pin     | B5  | B4  | B3  | C3  |
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/k_numpad17:default
+
+Bootmagic is enabled.  Press the key at (0,0) while plugging the keyboard in to jump to bootloader. This is numlock in the default keymap.
+
+Keyboard Maintainer: [Karlssn](https://github.com/Karlssn)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/readme.md]:
```suggestion
* Keyboard Maintainer: [Karlssn](https://github.com/Karlssn)
```

File: keyboards/handwired/k_numpad17/readme.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+Hardware: [Casefile](https://www.thingiverse.com/thing:4062409)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/readme.md]:
```suggestion
* Hardware Availability: [Casefile](https://www.thingiverse.com/thing:4062409)
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Key at 0,0 makes the keyboard go into bootloader
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no	# Mouse keys
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = yes	# Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes	# Console for debug
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Key at 0,0 makes the keyboard go into bootloader
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes	# Audio control and System control
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+CONSOLE_ENABLE = yes	# Console for debug(+400)
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality (+4870)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality
```

File: keyboards/handwired/k_numpad17/rules.mk [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality (+1150)
```
Comment (fauxpark) [inline keyboards/handwired/k_numpad17/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**